estimagic/optimization/nlopt_optimizers.py

```diff
@@ -141,6 +140,899 @@ def nlopt_neldermead(
+
+    - convergence.relative_params_tolerance (float):  Stop when the relative movement
+      between parameter vectors is smaller than this.
+    - convergence.relative_criterion_tolerance (float): Stop when the relative
+      improvement between two iterations is smaller than this.
+    - stopping_max_criterion_evaluations (int): If the maximum number of function
+      evaluation is reached, the optimization stops but we do not count this
+      as convergence.
+
+    """
+    out = _minimize_nlopt(
+        criterion_and_derivative,
+        x,
+        lower_bounds,
+        upper_bounds,
+        algorithm=nlopt.LD_MMA,
+        algorithm_name="nlopt_mma",
+        convergence_xtol_rel=convergence_relative_params_tolerance,
+        convergence_xtol_abs=convergence_absolute_params_tolerance,
+        convergence_ftol_rel=convergence_relative_criterion_tolerance,
+        convergence_ftol_abs=convergence_absolute_criterion_tolerance,
+        stopping_max_eval=stopping_max_criterion_evaluations,
+    )
+
+    return out
+
+
+def nlopt_var(
+    criterion_and_derivative,
+    x,
+    lower_bounds,
+    upper_bounds,
+    *,
+    convergence_relative_params_tolerance=CONVERGENCE_RELATIVE_PARAMS_TOLERANCE,
+    convergence_absolute_params_tolerance=CONVERGENCE_ABSOLUTE_PARAMS_TOLERANCE,
+    convergence_relative_criterion_tolerance=CONVERGENCE_RELATIVE_CRITERION_TOLERANCE,
+    convergence_absolute_criterion_tolerance=CONVERGENCE_ABSOLUTE_CRITERION_TOLERANCE,
+    stopping_max_criterion_evaluations=STOPPING_MAX_CRITERION_EVALUATIONS,
+    rank_variant=1,
+):
+
+    """Minimize a scalar function limited memory switching variable-metric method.
+
+    The algorithm relies on saving only limited number M of past updates of the
+    gradient to approximate the inverse hessian. The large is M, the more memory is
+    consumed
+
+    Detailed explanation of the algorithm, including its two variations of  rank-2 and
+    rank-1 methods can be found in the following paper:
+    J. Vlcek and L. Luksan, "Shifted limited-memory variable metric methods for
+    large-scale unconstrained minimization," J. Computational Appl. Math. 186,
+    p. 365-390 (2006).
+
+    ``nlopt_svmm`` supports the following ``algo_options``:
+
+    - convergence.relative_params_tolerance (float):  Stop when the relative movement
+      between parameter vectors is smaller than this.
+    - convergence.relative_criterion_tolerance (float): Stop when the relative
+      improvement between two iterations is smaller than this.
+    - stopping_max_criterion_evaluations (int): If the maximum number of function
+      evaluation is reached, the optimization stops but we do not count this
+      as convergence.
+
+    """
+    if rank_variant == 1:
+        algo = nlopt.LD_VAR1
+    elif rank_variant == 2:
+        algo = nlopt.LD_VAR2
+    else:
+        raise ValueError(
+            "nlopt supports only rank-1 and rank-2 methods of shifting variable-"
```

Comment:
You need to tell the user more precisely what to do. In the current implementation, she might have passed "rank-1" and get an error that only "rank-1" and "rank-2" is supported. Ideally, error messages first tell you what is possible and then tell you what you did wrong and how you can fix it.